validator/validation/validation.py

```diff
@@ -58,6 +59,21 @@ def _get_actual_time_range(val_run, dataset_version_id):
 
     return [actual_start, actual_end]
 
+def _get_reference_reader(val_run):
+    ref_reader = create_reader(val_run.reference_configuration.dataset, val_run.reference_configuration.version)
+
+    # we do the dance with the filtering below because filter may actually change the original reader, see ismn network selection
+    ref_reader = setup_filtering(ref_reader, list(val_run.reference_configuration.filters.all()),\
+                                 list(val_run.reference_configuration.parametrisedfilter_set.all()),\
+                                 val_run.reference_configuration.dataset, val_run.reference_configuration.variable)
+    while(hasattr(ref_reader, 'cls')):
+        ref_reader = ref_reader.cls
```

Comment:
Could you maybe add a comment about what you are doing here and why it is necessary? This is not clear to me.

Reply:
This (and the one above) are actually not my changes in the code. I'm afraid I can't explain it without digging in the code

Reply:
I see, it's already quite old. Then just leave it, and we hope it never breaks ;)

Reply:
As far as I can see it tries to find the original reader without adapters.

Reply:
@wpreimes do you know what's happening here?

Reply:
it takes the reader and repeats calling the base class (cls) until it reaches the base reader

Reply:
I think the while is necessary because we dont know how many adapters are around the base reader
``AdaptedReader(AdaptedReader(...(Reader())..)``